pkg/kvstore/consul.go

```diff
@@ -407,3 +453,135 @@ func (c *ConsulClient) DeleteTree(path string) error {
 	_, err := c.Client.KV().DeleteTree(path, nil)
 	return err
 }
+
+// Set sets value of key
+func (c *ConsulClient) Set(key string, value []byte) error {
+	_, err := c.KV().Put(&consulAPI.KVPair{Key: key, Value: value}, nil)
+	return err
+}
+
+// Delete deletes a key
+func (c *ConsulClient) Delete(key string) error {
+	_, err := c.KV().Delete(key, nil)
+	return err
+}
+
+// Get returns value of key
+func (c *ConsulClient) Get(key string) ([]byte, error) {
+	pair, _, err := c.KV().Get(key, nil)
+	if err != nil {
+		return nil, err
+	}
+	if pair == nil {
+		return nil, nil
+	}
+	return pair.Value, nil
+}
+
+// GetPrefix returns the first key which matches the prefix
+func (c *ConsulClient) GetPrefix(prefix string) ([]byte, error) {
+	pairs, _, err := c.KV().List(prefix, nil)
```

Comment:
You give no option. You may want to consider the RequireConsistent option to true. This will ensure consistent read semantics. This would be consistent with how you query etcd. In your etcd kvstore implementation, you don't specify WithSerializable, so your etcd queries have the default linearizable semantics, equivalent to Consul's RequireConsistent.